src/boot/efi/random-seed.c

```diff
@@ -221,31 +117,76 @@ static void validate_sha256(void) {
 }
 
 EFI_STATUS process_random_seed(EFI_FILE *root_dir, RandomSeedMode mode) {
-        _cleanup_free_ void *seed = NULL, *new_seed = NULL, *rng = NULL, *for_kernel = NULL, *system_token = NULL;
+        _cleanup_erase_ unsigned char random_bytes[DESIRED_SEED_SIZE], hash_key[HASH_VALUE_SIZE];
+        struct linux_efi_random_seed *prior_seed_table = NULL, *new_seed_table;
```

Comment:
Please mark `new_seed_table` as `_cleanup_free_`. `FreePool` doesn't care what memory type it is, as long as it's from `AllocatePool`.

Reply:
Ooolala fancy rust semantics. Okay, will do.

Reply:
Done. (Will push shortly.)